### PR TITLE
15-12-2020

### PR DESCRIPTION
**Pull request by DaliaZhran**

### CODE REVIEW

File: 11. Container With Most Water.py [hunk 1]
```diff
@@ -7,7 +7,27 @@
 
 """
```
Comment (EhabIbrahimH) [inline 11. Container With Most Water.py:9]: Check also https://leetcode.com/problems/trapping-rain-water/ with DP and 2 pointers approaches.

File: 3Sum Smaller.py [hunk 2]
```diff
@@ -8,11 +8,40 @@
```
Comment (EhabIbrahimH) [inline 3Sum Smaller.py:8]: This problem is more tricky at the interviews than when you see the solution, make sure you totally understand it and can code it without help

File: Multiply Strings.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        :type num2: str
+        :rtype: str
+        """
+        result = [0] * (len(num1) + len(num2))
+        pos = len(result) - 1
+
+        for i in range(len(num1) - 1, -1, -1):
+            temp_pos = pos
+            for j in range(len(num2) - 1, -1, -1):
+                result[temp_pos] += int(num1[i]) * int(num2[j])
```
Comment (EhabIbrahimH) [inline Multiply Strings.py:29]: This mustn't be used, resolve without any casting

File: Rotate Image.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        :rtype: None Do not return anything, modify matrix in-place instead.
+        """
+        n = len(matrix[0])
+        for i in range(n):
+            for j in range(n - 1, i - 1, -1):
+                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
+
+        # reverse the rows
+        for i in range(n):
+            matrix[i] = reversed(matrix[i])
```
Comment (EhabIbrahimH) [inline Rotate Image.py:31]: This is cool to mention that you new reversed built in function, but you must know how to do it optimally without built in functions

File: Rotate Image.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        :type matrix: List[List[int]]
+        :rtype: None Do not return anything, modify matrix in-place instead.
+        """
+        n = len(matrix)
+        for i in range(n):
+            for j in range(i):
+                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
+
+        # reverse columns
+        for i in range(n / 2):
```
Comment (EhabIbrahimH) [inline Rotate Image.py:61]: If you get problem like this, mention the reverse built in function and this one

**Review by EhabIbrahimH — CHANGES_REQUESTED** [review]
Good job, waiting for more

File: 11. Container With Most Water.py (hunk 1 above)
Comment (DaliaZhran) [reply 11. Container With Most Water.py:9]: done

**Pull request merged**